File: android/app/src/main/java/com/tomato/TalkingDataModule.java
package com.tomato;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.tendcloud.tenddata.TCAgent;
import com.tendcloud.tenddata.TDAccount;

import java.util.Map;

/**
 * Created by eddie on 13/9/2017.
 */

public class TalkingDataModule extends ReactContextBaseJavaModule {

    public TalkingDataModule(ReactApplicationContext reactContext) {
        super(reactContext);
    }

    @ReactMethod
    public void onEvent(String eventID) {
        TCAgent.onEvent(getReactApplicationContext(), eventID);
    }

    @ReactMethod
    public void onEventWithLabel(String eventID, String label) {
        TCAgent.onEvent(getReactApplicationContext(), eventID, label);
    }

    @ReactMethod
    public void onEventWithLabelAndParams(String eventID, String label, Map params) {
        TCAgent.onEvent(getReactApplicationContext(), eventID, label, params);
    }

    @ReactMethod
    public void onRegister(int type, String accountID, String accountName) {
        TDAccount.AccountType accountType = TDAccount.AccountType.values()[type];
        TCAgent.onRegister(accountID, accountType, accountName);
    }

    @ReactMethod
    public void onLogin(int type, String accountID, String accountName) {
        TDAccount.AccountType accountType = TDAccount.AccountType.values()[type];
        TCAgent.onLogin(accountID, accountType, accountName);
    }

    @ReactMethod
    public void trackPageBegin(String pageName) {
        TCAgent.onPageStart(getReactApplicationContext(), pageName);
    }

    @ReactMethod
    public void trackPageEnd(String pageName) {
        TCAgent.onPageEnd(getReactApplicationContext(), pageName);
    }

    @Override
    public String getName() {
        return "TalkingDataModule";
    }
}

File: ios/TalkingDataModule.m
//
//  TalkingDataModule.m
//  Tomato
//
//  Created by Eddie Zhou on 7/9/2017.
//  Copyright © 2017 Facebook. All rights reserved.
//

#import "TalkingDataModule.h"
#import "TalkingData.h"

@implementation TalkingDataModule

RCT_EXPORT_MODULE();

RCT_EXPORT_METHOD(onEvent:(NSString *)eventID){
  [TalkingData trackEvent:eventID];
}

RCT_EXPORT_METHOD(onEventWithLabel:(NSString *)eventID EvnetLabel:(NSString *)EvnetLabel){
  [TalkingData trackEvent:eventID label:EvnetLabel];
}

RCT_EXPORT_METHOD(onEventWithLabelAndParams:(NSString *)eventID EvnetLabel:(NSString *)EvnetLabel parameters:(NSDictionary *)parameters){
  [TalkingData trackEvent:eventID label:EvnetLabel parameters:parameters];
}

RCT_EXPORT_METHOD(onRegister:(NSUInteger)type accountID:(NSString *)accountID accountName:(NSString *)accountName){
  [TalkingData onRegister:accountID type:type name:accountName];
}

RCT_EXPORT_METHOD(onLogin:(NSUInteger)type accountID:(NSString *)accountID accountName:(NSString *)accountName){
  [TalkingData onLogin:accountID type:type name:accountName];
}

RCT_EXPORT_METHOD(trackPageBegin:(NSString *)pageName){
  [TalkingData trackPageBegin:pageName];
}

RCT_EXPORT_METHOD(trackPageEnd:(NSString *)pageName){
  [TalkingData trackPageEnd:pageName];
}

@end
